ground_truth_labeling_jobs/ground_truth_conversion_scripts/GT_semantic_segmentation_to_COCO.ipynb

```diff
@@ -0,0 +1,443 @@
+    "    seg_img = np.uint8(seg_img)\n",
+    "\n",
+    "    f_bmask = np.asfortranarray(seg_img)\n",
+    "    encoded_GT = mask.encode(f_bmask)\n",
+    "    area_GT = mask.area(encoded_GT)\n",
+    "    bb_GT = mask.toBbox(encoded_GT)\n",
+    "    contours = measure.find_contours(seg_img, 0.5)\n",
+    "    annotation = {\n",
+    "            \"category_id\": category_id,\n",
+    "            \"id\": idd,\n",
+    "            \"image_id\": image_id,\n",
+    "            \"iscrowd\": 0,\n",
+    "            \"segmentation\": [],\n",
+    "            \"area\": area_GT.tolist(),\n",
+    "            \"bbox\": bb_GT.tolist(),\n",
+    "        }\n",
+    "\n",
+    "    for contour in contours:\n",
+    "        contour = np.flip(contour, axis=1)\n",
+    "        segmentation = contour.ravel().tolist()\n",
+    "        annotation[\"segmentation\"].append(segmentation)\n",
+    "    return annotation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Download the mask output from S3"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "import time\n",
+    "annotations = []\n",
+    "\n",
+    "\n",
+    "path = \"/\".join(data_objs[0][JOBNAME+'-ref'].split(\"/\")[:-1]) + \"/\"\n",
+    "!aws s3 cp --recursive $path ./output_mask/"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate segmentation data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "\n",
+    "import matplotlib\n",
+    "from skimage import io\n",
+    "from skimage.viewer import ImageViewer\n",
+    "import matplotlib.pyplot as plt\n",
+    "from skimage.color import rgb2gray\n",
+    "\n",
+    "\n",
+    "idd = 0\n",
+    "all_annotations = []\n",
+    "annotations = []\n",
+    "for image_id, file in enumerate(data_objs):\n",
+    "    \n",
+    "    img_file = 'output_mask/'+ntpath.basename(file[JOBNAME + '-ref']) #subdir + os.sep + file\n",
+    "    print(img_file)\n",
+    "    a = io.imread(img_file,plugin='matplotlib')\n",
+    "    colors = np.unique(a.reshape(-1, a.shape[2]), axis=0)\n",
+    "    segments = []\n",
+    "    for i in range(colors.shape[0]):\n",
+    "        if colors[i][:-1].mean() < 1:\n",
+    "            segments.append(colors[i])\n",
+    "    \n",
+    "    for idds, seg in enumerate(range(len(segments))):\n",
+    "\n",
+    "        color_hex = matplotlib.colors.to_hex(segments[seg])\n",
+    "        category_id = mapping[color_hex]\n",
+    "        img = io.imread(img_file, plugin='matplotlib')\n",
+    "        msk = segments[seg]  \n",
+    "        masked_img = cv2.inRange(img, msk, msk)\n",
+    "        \n",
+    "        if len(segments) > 1:\n",
+    "            img[masked_img>0]=(255,255,255)\n",
```

Comment:
When I try to run this, this breaks, as img is a 4-channel (RGBA) mask. Maybe you want to do
```suggestion
    "            img[masked_img>0]=(255,255,255,255)\n",
```
instead?

Reply:
@natsirtguy What error do you get? It works fine for me with no issues.

Reply:
Still getting it:
```
<ipython-input-8-ba93f71864dd> in <module>
     24         masked_img = cv2.inRange(img, msk, msk)
     25         if len(segments) > 1:
---> 26             img[masked_img>0]=(255,255,255)
     27         annot = generate_segmentation(img,category_id, idd, image_id)
     28         idd+=1

ValueError: shape mismatch: value array of shape (3,) could not be broadcast to indexing result of shape (1764,4)
```
It's possible that the output format changed from when you ran your labeling job. Did you use one annotator per image for yours?

Reply:
Fixed